tests/integration/adapter-tests.js

```diff
@@ -26,6 +26,19 @@ export default function adapterTests() {
     expect(res.status).to.equal(404);
   });
 
+  it('should handle 204s with empty response body', async function() {
+    let res = await this.fetchRecord({
+      method: 'POST',
+      body: `''`
```

Comment:
Actually in doing this, I think the bug is elsewhere.  I shouldn't need to pass an empty string here to mimic the bug.